src/site/content/en/blog/sign-up-form-best-practices/index.md

```diff
@@ -0,0 +1,418 @@
+     <source src="https://samdutton.com/medium-sign-in.mp4" type="video/mp4">
+   </video>
+  <figcaption class="w-figcaption">Password-free sign-in on medium.com.</figcaption>
+</figure>
+
+As with federated login, this has the added benefit that you don't have to manage user passwords. 
+
+## Consider session length {: #session-length}
+ 
+Whatever approach you take to user identity. you'll need to make a careful decision about session 
+length: how long the user remains logged in, and what might cause you to log them out.
+
+There's no one hard-and-fast rule, but you need to consider mobile versus desktop, and whether 
+users are sharing on desktop, or sharing devices. 
+
+{% Aside %}
+You can get around some of the issues of shared devices by enforcing re-authentication for sensitive 
+features, for example when a purchase is made or an account updated. You can find out more about 
+ways to implement re-authentication from the codelab [Your First WebAuthn App](https://codelabs.developers.google.com/codelabs/webauthn-reauth/#0).
+{% endAside %}
+
+## Help password managers securely suggest and store passwords {: #help-password-managers}
+
+You can help third party and built-in browser password managers suggest and store passwords, so that 
+users don't need to choose, remember or type passwords themselves. Password managers work well in 
+modern browsers, synchronizing accounts across devices, across native and web apps—and for new 
+devices. 
+
+This makes it extremely important to code sign-up forms correctly, in particular to use the correct 
+autocomplete values. For sign-up forms use `autocomplete="new-password"` for new passwords, and add 
+correct autocomplete values to other form fields wherever possible, such as `autocomplete="email"` 
+and `autocomplete="tel"`.
+
+You should also use the appropriate [`type` attribute](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/email) 
+to provide the right keyboard on mobile and enable basic built-in validation by the browser. 
+You can find out more from [Payment and address form best practices](/payment-and-address-form-best-practices#type).
+
+{% Aside %}
+[Sign-in form best practices](/sign-in-form-best-practices) has lots more tips on how to improve 
+form design, layout and accessibility, and how to code forms correctly in order to take advantage 
+of built-in browser features.
+{% endAside %}
+
+## Ensure users enter secure passwords {: #secure-passwords}
+
+Enabling password managers to suggest passwords is the best option, and you should encourage 
+users to accept the strong passwords suggested by browsers and third-party browser managers. 
+
+However, many users want to enter their own passwords, so you need to implement rules for password 
+strength. [The US National Institute of Standards and Technology](https://pages.nist.gov/800-63-3/sp800-63b.html#5-authenticator-and-verifier-requirements) explains 
+how to avoid insecure passwords.
```

Comment:
```suggestion
However, many users want to enter their own passwords, so you need to implement rules for password 
strength. The US National Institute of Standards and Technology explains 
[how to avoid insecure passwords](https://pages.nist.gov/800-63-3/sp800-63b.html#5-authenticator-and-verifier-requirements).
```